src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -57,43 +61,66 @@ abstract class ListDatabaseRequestHandler<T> : IListDatabaseRequestHandler
     {
         private static readonly string[] SystemDatabases = new string[] { "master", "model", "msdb", "tempdb" };
 
-        public abstract string QueryText { get; }
-
         public ListDatabasesResponse HandleRequest(ISqlConnectionFactory connectionFactory, ConnectionInfo connectionInfo)
         {
             ConnectionDetails connectionDetails = connectionInfo.ConnectionDetails.Clone();
+            var serverConnection = ConnectionService.OpenServerConnection(connectionInfo);
+
+            SMO.Server SQLServer = new SMO.Server(serverConnection);
+
 
-            // Connect to master
-            connectionDetails.DatabaseName = "master";
-            using (var connection = connectionFactory.CreateSqlConnection(ConnectionService.BuildConnectionString(connectionDetails), connectionDetails.AzureAccountToken))
+            List<T> result = new List<T>();
+            ListDatabasesResponse response = new ListDatabasesResponse();
+
+
+            foreach (SMO.Database db in SQLServer.Databases)
```

Comment:
@shueybubbles Remind me - does SMO already filter out inaccessible DBs from the DB list for a server?

Reply:
I did check it out. From my testing, it returned all the databases that are present in sys.databases for the connection.

Reply:
SMO has some exception handlers in place to handle failure to fetch the Edition of each database in the list, so it sets Edition to Unknown on those it can't login to, but it has no way of excluding such DBs from the list.